syntax = "proto3";

package opcua;

// OPC-UA Bronze Record Schema
// Matches the Databricks Unity Catalog table schema
message OPCUABronzeRecord {
    // Timestamps in microseconds (int64 for large values)
    int64 event_time = 1;
    int64 ingest_time = 2;

    // Source information
    string source_name = 3;
    string endpoint = 4;

    // OPC-UA specific fields
    int32 namespace = 5;
    string node_id = 6;
    string browse_path = 7;

    // Status information
    int64 status_code = 8;
    string status = 9;

    // Value information
    string value_type = 10;
    string value = 11;
    double value_num = 12;
    bytes raw = 13;

    // PLC information (optional)
    string plc_name = 14;
    string plc_vendor = 15;
    string plc_model = 16;
}
